code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -786,12 +825,49 @@ GLOBAL_LIST_INIT(ai_verbs_default, list(
 
 	Bot.call_bot(src, waypoint)
 
+/mob/living/silicon/ai/alarm_triggered(src, class, area/A, list/O, obj/alarmsource)
+	if(!(class in alarms_listend_for))
+		return
+	if(alarmsource.z != z)
+		return
+	if(stat == DEAD)
+		return TRUE
+	if(O)
+		var/obj/machinery/camera/C = locateUID(O[1])
+		if(O.len == 1 && !QDELETED(C) && C.can_use())
```

Comment:
length.
Can also cache it since it's used more often than once. Yet not that big a deal